spec/controllers/equipment_items_controller_spec.rb

```diff
@@ -1,348 +1,261 @@
+          EquipmentItemMock.new(traits: [:findable], equipment_model: model)
+        end
+        before { put :deactivate, id: item.id, **opts }
+        it { is_expected.to set_flash[flash_type] }
+        it { is_expected.to redirect_to(model) }
+        it 'does not deactivate the item' do
+          expect(item).not_to have_received(:deactivate)
+        end
+      end
+      it_behaves_like 'unsuccessful', :notice, deactivation_cancelled: true
+      it_behaves_like 'unsuccessful', :notice, deactivation_cancelled: true,
+                                               deactivation_reason: 'reason'
+      it_behaves_like 'unsuccessful', :error
+      context 'successful' do
+        let!(:item) { EquipmentItemMock.new(traits: [:findable]) }
+        before do
+          request.env['HTTP_REFERER'] = '/referrer'
+          allow(item).to receive(:current_reservation).and_return(false)
+          put :deactivate, id: item.id, deactivation_reason: 'reason'
+        end
+        it { is_expected.to redirect_to('/referrer') }
+        it 'deactivates the item' do
+          expect(item).to have_received(:deactivate)
+            .with(hash_including(:user, :reason))
+        end
       end
-      it { expect(response).to be_redirect }
-      it { expect(item.deactivation_reason).to eq('Because I can') }
-      it { expect(item.deleted_at).not_to be_nil }
-      it 'should change the notes' do
-        new_item = FactoryGirl.create(:equipment_item)
-        put :deactivate, id: new_item, deactivation_reason: 'reason'
-        expect { new_item.reload }.to change(new_item, :notes)
+      context 'with reservation' do
+        let!(:item) { EquipmentItemMock.new(traits: [:findable]) }
+        let!(:res) { ReservationMock.new }
+        before do
+          request.env['HTTP_REFERER'] = '/referrer'
+          allow(item).to receive(:current_reservation).and_return(res)
+          put :deactivate, id: item.id, deactivation_reason: 'reason'
+        end
+        it 'archives the reservation' do
+          expect(res).to have_received(:archive)
+        end
       end
     end
-
-    context 'with non-admin user' do
+    context 'with normal user' do
       before do
-        sign_in FactoryGirl.create(:user)
-      end
-      it 'should redirect to root' do
-        put :deactivate, id: item, deactivation_reason: "Because I can't"
-        expect(response).to redirect_to(root_url)
+        mock_user_sign_in
+        put :deactivate, id: 1, deactivation_reason: 'reason'
       end
+      it_behaves_like 'redirected request'
     end
   end
 
   describe 'PUT activate' do
-    before { request.env['HTTP_REFERER'] = '/referrer' }
     context 'with admin user' do
+      let!(:item) { EquipmentItemMock.new(traits: [:findable], notes: '') }
       before do
-        sign_in FactoryGirl.create(:admin)
-        put :activate, id: deactivated_item
-        deactivated_item.reload
+        mock_user_sign_in(UserMock.new(:admin, md_link: 'admin'))
```

Comment:
it gets called in the action in order to add the user's info to the item's notes